test/index.js

```diff
@@ -1,81 +1,97 @@
-function throws(f, type, message) {
-  assert.throws(f, function(err) {
-    return err.constructor === type && err.message === message;
-  });
+//    eq :: (a, b) -> Undefined !
+function eq(actual, expected) {
+  assert.strictEqual (arguments.length, eq.length);
+  assert.strictEqual (Z.toString (actual), Z.toString (expected));
+  assert.strictEqual (Z.equals (actual, expected), true);
 }
 
-//  version :: String
-var version = '0.14.0';  // updated programmatically
-
+//    notImplemented :: () -> Undefined !
+const notImplemented = () => { throw new Error ('Not implemented'); };
 
-var def = $.create({checkTypes: true, env: $.env});
-
-var a = $.TypeVariable('a');
-var b = $.TypeVariable('b');
-var c = $.TypeVariable('c');
-var d = $.TypeVariable('d');
-var m = $.UnaryTypeVariable('m');
-
-function list() { return Array.prototype.slice.call(arguments); }
-
-var $0 = def('$0', {}, [$.Array(a)], list);
-var $1 = def('$1', {}, [a, $.Array(a)], list);
-var $2 = def('$2', {}, [a, a, $.Array(a)], list);
-var $3 = def('$3', {}, [a, a, a, $.Array(a)], list);
-var $4 = def('$4', {}, [a, a, a, a, $.Array(a)], list);
-var $5 = def('$5', {}, [a, a, a, a, a, $.Array(a)], list);
-var $6 = def('$6', {}, [a, a, a, a, a, a, $.Array(a)], list);
-var $7 = def('$7', {}, [a, a, a, a, a, a, a, $.Array(a)], list);
-var $8 = def('$8', {}, [a, a, a, a, a, a, a, a, $.Array(a)], list);
-var $9 = def('$9', {}, [a, a, a, a, a, a, a, a, a, $.Array(a)], list);
-
-
-var MAX_SAFE_INTEGER = Math.pow(2, 53) - 1;
-var MIN_SAFE_INTEGER = -MAX_SAFE_INTEGER;
+//    throws :: (Function, TypeRep a, String) -> Undefined
+const throws = (thunk, type, message) => {
+  assert.throws (thunk,
+                 err => err.constructor === type && err.message === message);
+};
 
-//  Integer :: Type
-var Integer = $.NullaryType(
-  'my-package/Integer',
-  'http://example.com/my-package#Integer',
-  function(x) {
-    return $.Number._test(x) &&
-           Math.floor(x) === Number(x) &&
-           x >= MIN_SAFE_INTEGER &&
-           x <= MAX_SAFE_INTEGER;
-  }
-);
+//    version :: String
+const version = '0.14.0';  // updated programmatically
+
+
+const def = $.create ({checkTypes: true, env: $.env});
+
+const a = $.TypeVariable ('a');
+const b = $.TypeVariable ('b');
+const c = $.TypeVariable ('c');
+const d = $.TypeVariable ('d');
+const m = $.UnaryTypeVariable ('m');
+
+//    $0 :: () -> Array a
+const $0 =
+def ('$0')
+    ({})
+    ([$.Array (a)])
+    (() => []);
+
+//    $1 :: a -> Array a
+const $1 =
+def ('$1')
+    ({})
+    ([a, $.Array (a)])
+    (x => [x]);
+
+//    $2 :: a -> a -> Array a
+const $2 =
+def ('$2')
+    ({})
+    ([a, a, $.Array (a)])
+    (x => y => [x, y]);
+
+//    $3 :: a -> a -> a -> Array a
+const $3 =
+def ('$3')
+    ({})
+    ([a, a, a, $.Array (a)])
+    (x => y => z => [x, y, z]);
+
+//    $26 :: a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> a -> Array a
```

Comment:
Haha

Reply:
I wanted to show that we're no longer limited to nine parameters. :joy: